### ISSUE TITLE
False positive lamp post with waste bin

### ISSUE THREAD

**Issue by Venefilyn** [osm-fr/osmose-backend#2243]
osmose needs to be more refined when a node has `amenity=waste_basket` and `highway=street_lamp`

It's fairly used all around and you see waste baskets attach to street lamps all over the place

The error Osmose reports is that it's unsual to use `amenity=` and `highway=` together

**Comment by Famlam** [comment]
Shouldn't this be [`bin=yes`](https://wiki.openstreetmap.org/wiki/Key:bin), if you're not mapping them separately? Otherwise you're not mapping according to [one feature, one element](https://wiki.openstreetmap.org/wiki/One_feature,_one_OSM_element).

For instance, add colour=yellow, and tell me if this is the color of the bin or the lamp post. On the other hand, bin=yes only says there's a bin around, not that the lamp post "is" also the bin.

**Comment by Venefilyn** [comment]
To me it doesn't make sense that we have two different systems like this. Given that `bin=yes` can be on it's own as well without anything else it doesn't help things.

From what I gathered, for 99% of all items they will be tagged as `bin=yes` if it's on a transit platform as most apps suggest that. But I haven't seen many that are attached to things
https://taginfo.openstreetmap.org/compare/bin=yes/amenity=waste_basket

Even if you were to add `colour=yellow` when an item has a `bin=yes` you wouldn't be able to accurately figure out what is being coloured

Wonder if this is just a case for a proposal, the whole situation irks me :thinking:

**Comment by 501Ghost** [comment]
I recommend mapping the street lamps and waste bins separate. You can give the waste bins a `support=*` tag to indicate that they're attached to a lamp post.

**Comment by jajajaneeneenee** [comment]
I would also tag it as 2 separate nodes (amenity=waste_basket with support=street_lamp).

On a street lamp, also other things can be attached, for example guideposts (tourism=information, information=guidepost), emergency access points (highway=emergency_access_point or emergency=access_point), excrement_bags (vending=excrement_bags), .... And I already saw 3 or 4 different things attached to a street lamp ... 

Therefore, mapping as separate nodes is probably the only sensible option, even if this results in confusing "node stacks".

Unfortunately, OpenStreetMap lacks a really good tagging concept for such "multiple features" in one place.

**Comment by Famlam** [comment]
As fixing the "multiple features" issue of OSM is a bit out of the scope of Osmose I suspect this issue can be closed until OSM itself has found a solution?

**Comment by jajajaneeneenee** [comment]
I also think this could be closed ...

Only one thought: I don't know the exact text of the warning, sometimes the wording makes a difference (and a nicely worded advice that it is usually better to tag different features separately might be helpful).

**Comment by Famlam** [comment]
> Only one thought: I don't know the exact text of the warning, sometimes the wording makes a difference (and a nicely worded advice that it is usually better to tag different features separately might be helpful).

At this moment, the [warning message](https://github.com/osm-fr/osmose-backend/blob/master/plugins/TagRemove_Incompatibles.py#L31) is:
>Tag conflict
>Conflict between tags: amenity, highway

with the description in the side column:

>The object contains two incompatible tags.

We could maybe rephrase the latter to:

>This object has two tags that represent multiple features. According to the principle of [one feature, one OSM element](https://wiki.openstreetmap.org/wiki/One_feature,_one_OSM_element), these should be mapped as separate objects.

(Suggestions welcome. Note that the same analyser also warns for things you really won't find together, like a waterway and an aeroway)

**Comment by Venefilyn** [comment]
I think that would work, if it's unintended you would still be able to look at it and see what's wrong. 100% for rephrasing the message

**Comment by jajajaneeneenee** [comment]
Also 100% for rephrasing. Better wording can make a difference.

The text is fine I think and a good improvement. Another variant could be (but I am not a native English speaker)

_This object has two tags that represent different features. According to the principle of “one feature, one OSM element”, these should better be mapped as two separate objects._

And not sure if something like this should be added:

_(Even if it is exactly the same position and seems to be clear which tags belong to which feature.)_

Because that is probably the decision problem that has led many mappers (including me) to double tagging.

In addition, the principle “One feature, one OSM element” is not a really easy principle (only see the looooong text of the Wiki page).


Some other thoughts, maybe a little off-topic (but connected):

For me still a little confusing (after many years of mapping):
* Some cases seem to be accepted, like a (closedway) of a building with ONE shop or amenity (e.g. restaurant) or a hotel ...
* If you draw two openways or closedways with exact the same position (and the same nodes), e.g. landuse=meadow + barrier=fence, you can get another Osmose warning saying “two ways with exact the same position” (or similar). In any case, I have seen such warnings before, where I wondered how one should map it differently, and the principle “One feature, one OSM element” was also taken into account.
* Or if there is a barrier=retaining_wall with a fence on top, and you draw 2 ways with the same position, you also get such a warning I think – even if you added layer=1 to barrier=fence. (And you even couldn't map this with ONE way.)

I just want to point out that conflicting Osmosis warnings can occur, which is frustrating (and confusing) and no clear solution is provided. But again, it can be a reason for double tagging, because from a human perspective it is very often clear which tags belong to which feature (not always, of course), and double tagging seems to be the easiest way to go. 


But a first step could be a rephrasing here ...

**Comment by Famlam** [comment]
The updated text is live. (Translations will follow later, that's a manual synchronization)
